### DISCUSSION TITLE
SSH tunnel with Windows Teleport Connect at startup.

### DISCUSSION THREAD

**Discussion by sergefedorow** [gravitational/teleport#42668] [Q&A] [answered]
I want to have connection to remote host by https port 443 without additional application proxy and Teleport authentication. The cause is Windows webdav client does not allow me to login into Teleport. So, I have made a ssh tunnel like this: `tsh ssh --no-remote-exec --forward 443:localhost:443 root@remote-host`. It works fine. How to make it run at startup of Teleport Connect.exe? Is it possible to configure Windows Teleport Connect automatically open PowerShell terminal and execute predefined commands or script?

**Comment by stevenGravy** [comment, marked as answer]
Hi, this currently isn't an option.

**Comment by webvictim** [comment]
One thing you could do instead is to add the WebDAV target as a TCP application using Teleport application access - see docs here: https://goteleport.com/docs/application-access/guides/tcp/

Once completed, this would allow you to map a local TCP port to relay traffic to the WebDAV server inside Teleport Connect. This could be automatically restarted when Connect is opened and you log into the cluster - Connect can remember TCP ports linked to applications and reconnect them for you, but it can't do this with SSH port forwards.

**Comment by sergefedorow** [comment]
@webvictim thank you! I have made the tcp app and it works as charm. 
Only one thing, the Teleport Connect at startup always ask user to reopen previous sessions. 

![2024-08-02_16-32](https://github.com/user-attachments/assets/676b97ee-2d51-40b1-a200-c77a0eb55983)

This question annoying of the users. I did not find the property to force it automatically reconnect. There is no that kind of property in config file and in the command line of Teleport Connect.
Is it possible to reopen previous sessions by default?

**Reply by webvictim** [reply]
This isn't currently possible. Please raise a feature request in this repo if it's behaviour you'd like to see!

**Reply by zmb3** [reply]
Hmm, are you on macOS by chance? With Teleport VNet you shouldn't need to manage tunnels at all.

**Reply by sergefedorow** [reply]
@webvictim thanks, I'll do.
@zmb3 VNet is really a great thing, but for reasons beyond my comprehension, some users still use Windows. And by the professional technical reasons I'm on Linux. So, waiting for VNet for Windows and Linux.
